### PR TITLE
Fix Renesas Compiler

### PR DESCRIPTION
**Pull request by alanNz**
Fixed Renesas RX Family compiler to work with latest meson, updated cross-file, fixed assembly file use

### CODE REVIEW

**Comment by jpakkane** [comment]
As per [the documentation](https://mesonbuild.com/Contributing.html#keeping-pull-requests-up-to-date) please don't merge trunk to your branch. It should have only one commit with your new code.

**Comment by dcbaker** [comment]
@alanNz, if you do a `git rebase` on master, you should be able to easily get rid of the back merges, once those are taken care of this is ready to land.

**Comment by alanNz** [comment]
@dcbaker I've just rebased here and tested it, cheers

File: cross/ccrx.txt [hunk 1]
```diff
@@ -6,15 +6,53 @@ c = 'ccrx'
...
+  '-goptimize',
+  '-nostuff',
+  '-define=_REENTRANT,_FEVAL=0,_HAS_C9X_FAST_FMA=0,RX63_BOARD=CUSTOM631,__DBL8,DEVICE_PACKAGE_100_PIN,GHW_SINGLE_CHIP'
+  ]
+c_link_args = [
+  '-show=ALL',
+  '-list=device.map',
+  '-form=absolute',
+  '-nologo',
+  '-optimize',
```
Comment (jpakkane) [inline cross/ccrx.txt]: If this does the same as `-O2` does on GCC, it should not be listed in the cross file. All optimization and debug flags should come from the compiler class based on build settings, not from cross files.

Comment (alanNz) [reply cross/ccrx.txt]: good point changed now to less info

**Comment by jpakkane** [comment]
Thanks.

**Pull request merged**